### PR TITLE
[13.0][MIG] web_advanced_search: Migration to 13.0

### PR DESCRIPTION
**Pull request by Tardo**
As commented in https://github.com/OCA/web/issues/1517 this module is affected by https://github.com/odoo/odoo/issues/47950

cc @tecnativa TT19820

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
I miss the domain editor in debug mode: 
image

**Review by chienandalu — COMMENTED** [review]
Also field names in debug mode (surely the same issue):
image

**Comment by yajo** [comment]
@chienandalu AFAIK those are upstream features...

**Comment by chienandalu** [comment]
I tested the standard widget in filters and it was ok

File: web_advanced_search/static/src/js/web_advanced_search.js [hunk 1]
```diff
@@ -119,13 +115,19 @@ odoo.define("web_advanced_search", function(require) {
...
+            const proposition = new AdvancedSearchProposition(
                 this,
-                this.dataset.model,
+                this._modelName,
                 event.data.domain
             );
-            this.propositions = [proposition];
-            this._commitSearch();
+            // Necessary to ensure that the porposition have the 'fieldSelector'
+            // is filled
```
Comment (yajo) [inline web_advanced_search/static/src/js/web_advanced_search.js:124]:
```suggestion
            // Necessary to ensure that the proposition has the 'fieldSelector'
            // and it is filled
```

File: web_advanced_search/static/src/js/web_advanced_search.js [hunk 2]
```diff
@@ -119,13 +115,19 @@ odoo.define("web_advanced_search", function(require) {
...
+            _.defer(
+                function() {
+                    this.propositions = [proposition];
+                    this._commitSearch();
+                }.bind(this)
```
Comment (yajo) [inline web_advanced_search/static/src/js/web_advanced_search.js:129]: Better use an arrow function, which is bound by default:

```suggestion
                () => {
                    this.propositions = [proposition];
                    this._commitSearch();
                }
```

File: web_advanced_search/static/src/js/web_advanced_search.js [hunk 3]
```diff
@@ -283,7 +284,9 @@ odoo.define("web_advanced_search", function(require) {
                 return FieldManagerMixin._applyChanges.apply(this, arguments);
             }
 
-            return $.Deferred().resolve();
+            return new Promise(resolve => {
+                resolve();
+            });
```
Comment (yajo) [inline web_advanced_search/static/src/js/web_advanced_search.js:289]: I think this is not the same.

We were returning a resolved deferred before, so it should be something like:

```suggestion
            return new Promise().resolve();
```

File: web_advanced_search/static/src/js/human_domain.js [hunk 4]
```diff
@@ -1,69 +1,58 @@
...
-                value = _.str.sprintf('"%s"', this.value);
+            const chain = [];
+            let operator = this.operator_mapping[this.operator],
+                value = `"${this.value}"`;
             // Humanize chain
-            this.chain.split(".").forEach(function(element, index) {
+            const chain_splitted = this.chain.split(".");
+            const len = chain_splitted.length;
+            for (let x = 0; x < len; ++x) {
+                const element = chain_splitted[x];
```
Comment (yajo) [inline web_advanced_search/static/src/js/human_domain.js]: Is this worth it? Why not just:

```js
             this.chain.split(".").forEach((element, index) => {
```

Almost like before, but you avoid having to manually bind the function.

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by alexey-pelykh** [comment]
Just in case https://github.com/OCA/web/pull/1554 will be accepted and merged prior this one, it would be great to include in v13

**Comment by yajo** [comment]
On its way...

**Review by manuelcalerosolis — APPROVED** [review]
Functional review!

**Comment by pedrobaeza** [comment]
@Yajo is it OK for you?

@Tardo can you add #1554?

**Review by yajo — APPROVED** [review]

**Comment by oscars8a** [comment]
Hi everyone,

This PR correct actually the problem with advanced search filter?

I have rebuilt the runbot 1551 and I could prove that the problem persists.

Which one is the state of this problem for Odoo 13.0 and 12.0?
I have this problem in a Odoo 12.0 version.

I put a git that show the problem in the runbot 1551, and this is the link http://3465288-1551-0732a9.runbot2-3.odoo-community.org/

Thanks a lot.

Peek 2021-02-15 13-10

**Comment by pedrobaeza** [comment]
No, that problem is not notified anywhere AFAIK

**Pull request merged**